src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildAction.java

```diff
@@ -102,10 +102,24 @@ protected SauceCredentials getCredentials() {
      * @throws JSONException Not json returned properly
      */
     public static LinkedHashMap<String, JobInformation> retrieveJobIdsFromSauce(SauceREST sauceREST, Run build) throws JSONException {
-        SauceCredentials credentials = build.getAction(SauceOnDemandBuildAction.class).getCredentials();
+        SauceCredentials credentials = getBuildAction(build).getCredentials();
         return retrieveJobIdsFromSauce(sauceREST, build, credentials);
     }
 
+    /**
+     * @param build The build in progress
+     * @return the {@link SauceOnDemandBuildAction} instance which has been registered with the build
+     *         Can be null
+     */
+    private static SauceOnDemandBuildAction getBuildAction(Run build) {
+        SauceOnDemandBuildAction buildAction = build.getAction(SauceOnDemandBuildAction.class);
+        if (buildAction == null && build instanceof hudson.maven.MavenBuild) {
```

Comment:
Lets import MavenBuild class so its consistent

But yea, I can't wait till they remove MavenBuild. I'm not sure thats ever happening though.